### ISSUE TITLE
convert for Safari

### ISSUE THREAD

**Issue by kaiyuyue** [dralletje/Windowed#66]
**0. version**

  - Safari: 17.2.1
  - macOS: Sonoma 14.2.1
  - Xcode: 15.1

Just install the Xcode from the AppStore to prepare.

**1. git clone** 

`git clone https://github.com/dralletje/Windowed.git`

**2. modify some keys in `manifest.json`**

The `manifest.json` is located at the cloned folder `${path-to-project-folder}/Windowed/extension`.

  - `name`: remove all the blank space in it, for example, `"name": "Windowed",`
  - `developer`: delete the `developer` item and move this part to the description, for example, `"description": "developer: Michiel Dral (https://dral.eu/)",`
  - `applications`: just delete the item
 
**3. convert**

Run the following command in the terminal:

`xcrun safari-web-extension-converter ${path-to-project-folder}/Windowed/extension`

The output in the terminal:

```bash
Xcode Project Location: /Users/username
App Name: Windowed
App Bundle Identifier: com.yourCompany.Windowed
Platform: All
Language: Swift
```

After running the command successfully, then the Xcode will be automatically pop out for the project.

**4. build in Xcode**

<img width="1624" alt="Screenshot 2023-12-29 at 5 27 12 PM" src="https://github.com/dralletje/Windowed/assets/19852297/da29b384-2799-4b51-95e2-71f331676f84">

Then, click the `quit and open safari settings`

<img width="537" alt="Screenshot 2023-12-29 at 5 26 12 PM" src="https://github.com/dralletje/Windowed/assets/19852297/632d98b9-f7b8-40d7-bf0e-ed237ca2264c">

**5. open dev mode in safari**

<img width="948" alt="Screenshot 2023-12-29 at 5 29 35 PM" src="https://github.com/dralletje/Windowed/assets/19852297/89a76256-dbda-4571-824d-1fe6bd4101d5">

<img width="948" alt="Screenshot 2023-12-29 at 5 29 41 PM" src="https://github.com/dralletje/Windowed/assets/19852297/3df2c9be-e0f6-4bdb-a388-f368e60a838b">

**6. enable**

After enabling it in Safari, it should work as usual.

<img width="1624" alt="Screenshot 2023-12-29 at 5 31 39 PM" src="https://github.com/dralletje/Windowed/assets/19852297/77126e60-f5e9-4990-a163-ba4a29ee9460">

<img width="1624" alt="Screenshot 2023-12-29 at 5 36 01 PM" src="https://github.com/dralletje/Windowed/assets/19852297/376b1bd6-a10a-4cec-989e-7bdb2efd04ae">

**7. issue**

There are unsatisfied borders at the top and bottom.

<img width="1624" alt="Screenshot 2023-12-29 at 5 37 34 PM" src="https://github.com/dralletje/Windowed/assets/19852297/181c0c7f-462b-40cd-8e36-e964d9232d78">

Really would love to have it in Safari. Hope it will be considered for development for Safari.

**Comment by dralletje** [comment]
Woowwww that's so cool!
Thanks :)
Okay if I merge into this repo and publish with my Apple Developer account?
(I will change the logo and the "Pic-in-pic" text back through ;) )

**Comment by kaiyuyue** [comment]
Yes, definitely, please merge and distribute it via Apple Store. lol That will be the first and best Windowed extension for Safari.
